File: 0997-find-the-town-judge/0997-find-the-town-judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
      const  indeg =Array(n+1).fill(0);
         const out= Array(n+1).fill(0);
        //  for(const it of trust){
        //     out[it[0]]++;
        //     indeg[it[1]]++;
        //  }
          for(const [a,b] of trust){
            out[a]++;
            indeg[b]++;
         }
         for(let i=1;i<=n;i++){
            if(indeg[i]==n-1 && out[i]==0) return i;
         }
         return -1;
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
// o(n+m)
// m=trust relation size 
// n=number of nodes
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int>in(n+1);
         vector<int>out(n+1);
         for(auto it:trust){
            out[it[0]]++;
            in[it[1]]++;
         }
         for(int i=1;i<=n;i++){
            if(in[i]==n-1 && out[i]==0) return i;
         }
         return -1;
    }
};